lib/mayaUsd/ufe/Utils.h

```diff
@@ -231,5 +231,25 @@ Ufe::Selection recreateDescendants(const Ufe::Selection& src, const Ufe::Path& f
 MAYAUSD_CORE_PUBLIC
 std::vector<std::string> splitString(const std::string& str, const std::string& separators);
 
+class PullExtras
+{
+public:
+    void initRecursive(Ufe::SceneItem::Ptr);
+    void processItem(const Ufe::Path& path, const MObject& mayaObject);
+
+private:
+    std::unordered_map<Ufe::Path, MObject> _displayLayerMap;
```

Comment:
Why not store the MFnDisplayLayer so you don't need the extra check/object create in the processItem()?

Reply:
@seando-adsk I cannot put MFnDisplayLayer in a container because its operator= as well as its copy constructor are declared private and not accessible. Putting a pointer to MFnDisplayLayer is basically the same as using MObject